File: 2799-count-complete-subarrays-in-an-array/2799-count-complete-subarrays-in-an-array.cpp
class Solution {
public:
    int countCompleteSubarrays(vector<int>& A) {
        int n = A.size(), k = unordered_set<int>(A.begin(), A.end()).size(), res = 0, i = 0;
        unordered_map<int, int> count;
        for (int j = 0; j < n; ++j) {
            k -= count[A[j]]++ == 0;
            while (k == 0)
                k += --count[A[i++]] == 0;
            res += i;
        }
        return res;
    }
};

File: 2799-count-complete-subarrays-in-an-array/2799-count-complete-subarrays-in-an-array.java
class Solution {
   public int countCompleteSubarrays(int[] A) {
        Set<Integer> s = new HashSet<>();
        for (int a : A)
            s.add(a);
        int n = A.length, k = s.size(), res = 0, i = 0;
        Map<Integer, Integer> count = new HashMap<>();

        for (int j = 0; j < n; ++j) {
            if (count.getOrDefault(A[j], 0) == 0)
                k--;
            count.put(A[j], count.getOrDefault(A[j], 0) + 1);
            while (k == 0) {
                count.put(A[i], count.get(A[i]) - 1);
                if (count.get(A[i]) == 0)
                    k++;
                i++;
            }
            res += i;
        }
        return res;
    }
}